homeassistant/components/template/fan.py

```diff
@@ -316,17 +315,10 @@ async def async_oscillate(self, oscillating: bool) -> None:
         if self._set_oscillating_script is None:
             return
 
-        if oscillating in _VALID_OSC:
-            self._oscillating = oscillating
-            await self._set_oscillating_script.async_run(
-                {ATTR_OSCILLATING: oscillating}, context=self._context
-            )
-        else:
-            _LOGGER.error(
-                "Received invalid oscillating value: %s. Expected: %s",
-                oscillating,
-                ", ".join(_VALID_OSC),
-            )
+        self._oscillating = oscillating
+        await self._set_oscillating_script.async_run(
+            {ATTR_OSCILLATING: oscillating}, context=self._context
+        )
```

Comment:
`oscillating` is already a bool, so what is the purpose of checking if it is `in [True, False]`?

Reply:
Some legacy leftover probably.